### PR TITLE
Add chain config and CLI changes for Binance Smart Chain support

### PR DESCRIPTION
**Pull request by brendan-kelly**

### CODE REVIEW

File: turbo/stages/headerdownload/preverified_hashes_bsc_mainnet.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package headerdownload
+
+var bscMainnetPreverifiedHashes = []string{
```
Comment (AlexeyAkhunov) [inline turbo/stages/headerdownload/preverified_hashes_bsc_mainnet.go:3]: We should add those only once we have successfully synced to the BSC main net, we have a utility to generate those. Please remove for now

File: params/config.go [hunk 2]
```diff
@@ -476,7 +551,7 @@ func (c *ChainConfig) String() string {
 	default:
 		engine = "unknown"
 	}
-	return fmt.Sprintf("{ChainID: %v Homestead: %v DAO: %v DAOSupport: %v EIP150: %v EIP155: %v EIP158: %v Byzantium: %v Constantinople: %v Petersburg: %v Istanbul: %v, Muir Glacier: %v, Berlin: %v, London: %v, Arrow Glacier: %v, Engine: %v}",
+	return fmt.Sprintf("{ChainID: %v Homestead: %v DAO: %v DAOSupport: %v EIP150: %v EIP155: %v EIP158: %v Byzantium: %v Constantinople: %v Petersburg: %v Istanbul: %v , Muir Glacier: %v, Ramanujan: %v, Niels: %v, MirrorSync: %v, Berlin: %v, London: %v, Arrow Glacier: %v, Engine: %v}",
```
Comment (AlexeyAkhunov) [inline params/config.go:554]: Can you please print only Ethereum forks if BSC forks are all 0s, and print only BSC forks otherwise

File: go.sum [hunk 3]
```diff
@@ -1552,6 +1563,118 @@ honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc/go.mod h1:rf3lG4BRIbNafJWh
 honnef.co/go/tools v0.0.1-2019.2.3/go.mod h1:a3bituU0lyd329TUQxRnasdCoJDkEUEAqEt0JzvZhAg=
 honnef.co/go/tools v0.0.1-2020.1.3/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
 honnef.co/go/tools v0.0.1-2020.1.4/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
+lukechampine.com/uint128 v1.1.1/go.mod h1:c4eWIwlEGaxC/+H1VguhU4PHXNWDCDMUlWdIWl2j1gk=
```
Comment (AlexeyAkhunov) [inline go.sum:1566]: please run `go mod tidy`

**Comment by AlexeyAkhunov** [comment]
Thank you for the PR. Currently some tests are failing because for Ethereum, we check that forks have monotonically increasing numbers. You need to look into a solution to that, perhaps properly separating chain configuration Ethereum / BSC as we discussed

File: params/config.go (hunk 2 above)
Comment (brendan-kelly) [reply params/config.go:565]: Let me know if the new code is what will work

File: go.sum (hunk 3 above)
Comment (brendan-kelly) [reply go.sum:1566]: Running this did not clean out any extra dependencies

File: params/config.go (hunk 2 above)
Comment (AlexeyAkhunov) [reply params/config.go:565]: No, because for BSC test-nets it will still print Ethereum forks

Comment (AlexeyAkhunov) [reply params/config.go:565]: Check on consensus engine type - this should work

File: go.sum (hunk 3 above)
Comment (AlexeyAkhunov) [reply go.sum:1566]: All this stuff should not be here in the first place, you don't use any of these modules in this PR

Comment (brendan-kelly) [reply go.sum]: Cleaned up both `go.mod` and `go.sum` by checking out what is on `devel`

File: params/config.go (hunk 2 above)
Comment (brendan-kelly) [reply params/config.go]: Updated

**Pull request merged**